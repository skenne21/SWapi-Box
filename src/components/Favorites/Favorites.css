.Favs {
  position: absolute;
  top: 15px;
  right: 30px;
  margin: 0;
  width: 200px;
  background-color: #FFD700;
  border-radius: 30px;
}

.favs-button {
  width: 80%;
  height: 50px;
  margin-top: 15px;
  font-size: 1.4em;
}

.number {
  font-size: 2em;
  margin: 20px;
}