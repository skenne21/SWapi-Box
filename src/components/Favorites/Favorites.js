import React from 'react';
import PropTypes from 'prop-types';
import Button from '../Button/Button.js';


const Favorites = ({favorites, showFavorites}) => {
  const navButton = () => {
    return(
      <div>
        <Button
          name={'View Favorites'}
          controlFunc={showFavorites}
        />
        <p>{favorites.length}</p>
      </div>
    )
  }

  

  return (
    <div className='Favorites'>
      {navButton()}
    </div>
  ) 
}

Favorites.propTypes = {
  favorites: PropTypes.array,
  showFavorites: PropTypes.func
}

export default Favorites;

