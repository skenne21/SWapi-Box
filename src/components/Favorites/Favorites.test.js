import React from 'react';
import { shallow } from 'enzyme';
import Favorites from './Favorites.js';

describe('Favorites', () => {
  it('Should match the snapshot if the array is empty', () => {
    const mockFunction = jest.fn();
    const wrapper = shallow(
      <Favorites 
        favorites={[]}
        showFavorites={mockFunction}
      />);
    expect(wrapper).toMatchSnapshot();
  });
  
  it('Should match the snapshot if the array has Cards', () => {
    const card = {
      class: 'people',
      name: "Luke Skywalker",
      info: {
        species: "species: Human", 
        homeWorldName:  "homeworld: Tatooine", 
        population: "Population: 200000"
      }
    };
    const mockFunction = jest.fn();
    const wrapper = shallow(
      <Favorites 
        favorites={[card]}
        showFavorites={mockFunction}
      />);
    expect(wrapper).toMatchSnapshot();
  });
});