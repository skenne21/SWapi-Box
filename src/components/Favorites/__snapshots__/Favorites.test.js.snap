// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Favorites Should match the snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Favorites
    favorites={Array []}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <div>
        <Button
                controlFunc={undefined}
                name="View Favorites"
        />
        <p />
</div>,
      "className": "Favorites",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <Button
            controlFunc={undefined}
            name="View Favorites"
/>,
          <p />,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "controlFunc": undefined,
            "name": "View Favorites",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": 0,
          },
          "ref": null,
          "rendered": 0,
          "type": "p",
        },
      ],
      "type": "div",
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <div>
          <Button
                    controlFunc={undefined}
                    name="View Favorites"
          />
          <p />
</div>,
        "className": "Favorites",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <Button
              controlFunc={undefined}
              name="View Favorites"
/>,
            <p />,
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "controlFunc": undefined,
              "name": "View Favorites",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": 0,
            },
            "ref": null,
            "rendered": 0,
            "type": "p",
          },
        ],
        "type": "div",
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
